Module: Add server module

patrickelectric opened the pull request:
Add json and rest servers

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric committed:
Enable mavproxy to watch multiple messages

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric committed:
Module: Add restserver module

Add rest server

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

patrickelectric commented:
Just opening a review here before creating one in the original repository.

jaxxzer commented on MAVProxy/modules/mavproxy_server.py:
```diff
@@ -0,0 +1,235 @@
+        self.start()
+
+    def update_dict(self, mpstate):
+        '''We don't have time to wast'''
```
sp. waste

jaxxzer commented:
<s>This doesn't show up in `module list` output. Neither do depthoutput or gpsinput for that matter.</s> Sorted.

patrickelectric commented:
Are you running it on source or have you installed it ?
You can use this patch https://github.com/ArduPilot/MAVProxy/pull/504 to run without installing.

jaxxzer reviewed (changes requested):
For the jsonserver, Instead of requesting a specific message at a certain interval (this isn't even supported yet in the mavlink api), we should request a specific message, and push the messages out as soon as they are received from the autopilot. This can be done by hooking into (mavlink_packet)[https://github.com/bluerobotics/MAVProxy/blob/master/MAVProxy/modules/mavproxy_DepthOutput.py#L39]

This way updates are only pushed when there is new data, and as soon as it is available.

tridge committed:
added asterix -> ADSB module

tridge committed:
asterix: added track to squawk

tridge committed:
added support for more OBC DNFZ types

tridge committed:
wp: support adding a nofly zone

tridge committed:
added height labels to adsb objects

tridge committed:
genobstacles: added OBC 2018 obstacle generator

tridge committed:
adsb: use mavlink time

this allows for stopping with gdb

tridge committed:
asterix: use mavlink time

tridge committed:
genobstacles: limit plane bank angles to 45 degrees

tridge committed:
genobstacles: added stop option

peterbarker committed:
mavproxy_wp: summarily ignore mission requests not targeted at us

This squelches the warning "not loading waypoints" when working in a
multi-gcs environment

tridge committed:
asterix: cope with asterix msgs starting with (

peterbarker committed:
mavproxy_map: add popup menu option to remove nofly zones

tridge committed:
genobstacles: added drop and remove via map menu

tridge committed:
asterix: send pkts to all links

tridge committed:
obstacles: auto-set home location

peterbarker committed:
mavproxy_asterix: add filtering of packets sent to vehicles

peterbarker committed:
mavproxy_asterix: add bytes/second for adsb packets

tridge committed:
allow for multiple map modules to be loaded

tridge committed:
added map zoom, center and follow commands

peterbarker committed:
mavproxy_param.py: correct fetch-set in case logdir is None

tridge committed:
module: allow finding of modules by wildcard name

tridge committed:
wp: make movemulti work from other modules

tridge committed:
default to mavlink2

tridge committed:
fixed module unload error

tridge committed:
wp: clear loading_waypoints when done

tridge committed:
map: make secondary plane a blue fixed wing

tridge committed:
map: support saving and loading of map window layout

tridge committed:
layout: save console layout

tridge committed:
mp_image: support window layout save

tridge committed:
layout: enable by default

tridge committed:
layout: allow layout load when windows not instantiated

tridge committed:
layout: added debug code

tridge committed:
layout: prevent error on bad layout file

tridge committed:
map: fixed event handling

tridge committed:
layout: use .mavproxy for layout files

tridge committed:
genobstacles: use elevation map for ground height

tridge committed:
adsb: show OBC threat radius for all objects

tridge committed:
adsb: show obstacles on all maps

tridge committed:
don't use streamrate2 if streamrate1 is -1

tridge committed:
genobstacles: fixed duplicate obstacle drop

tridge committed:
asterix: added logging of all pkts

tridge committed:
map: fixed multi-map callbacks

tridge committed:
sensors: remove support for old GPS_RAW msg

tridge committed:
speech: allow listing of voices

peterbarker committed:
mavproxy_link: pass sysid-output position through to NMEA module

tridge committed:
asterix: default filter distance to 3km

tridge committed:
adsb: prevent incorrect timeout of adsb objects

tridge committed:
asterix: display asterix pkt count

tridge committed:
added module get_time() function

auto-scales for SITL with SIM_SPEEDUP

tridge committed:
asterix: improved time handling, and asterix filtering

tridge committed:
adsb: drop timeout to 5

tridge committed:
asterix: display initial asterix status

tridge committed:
asterix: fwd ADSB_VEHICLE pkts to sysid GCSs

tridge committed:
genobstacles: improved climb rate generation

tridge committed:
asterix: added in WGS84 to AMSL correction

tridge committed:
genobstacles: added WGS84 to AMSL conversion

tridge committed:
link: send GLOBAL_POSITION_INT to sysid outputs

this allows them to display retrieval aircraft pos

tridge committed:
asterix: added vertical filtering

tridge committed:
asterix: improved vertical filtering

tridge committed:
misseditor: fix jumps on row add/remove

tridge committed:
wp: fix jumps on wp remove/add

stephendade committed:
Win_Layout: window layout profiles on a per vehicle_name setting

tridge committed:
asterix: set squawk as timestamp in 0.1s increment

this allows for old and duplicate pkt detection on vehicle

tridge committed:
layout: fallback to name without vehicle

tridge committed:
map: display height of 2nd vehicle

tridge committed:
added environment variable substitution in scripts

tridge committed:
substitute: added script_fatal setting

tridge committed:
wp: suppress spurious errors

tridge committed:
map: show GPS pos, use yaw at low speed

tridge committed:
wp: faster nofly add

tridge committed:
genobstacles: fixed DropBird

peterbarker committed:
mavproxy_wp: add ability to save as human-readable CSV

peterbarker committed:
mavproxy_map: replace blue box to position target with a vector

tridge committed:
genobstacles: added ClearAll

tridge committed:
arm: check ICE disabled

tridge committed:
mp_image: supress exception

peterbarker committed:
mavproxy_graph: pretty-print some annoyingly-long field names

peterbarker committed:
mavproxy_map: verify lat/lon set before using to plot target vector

tridge committed:
graph: allow control of legend for graphs

tridge committed:
misc: make module public

tridge committed:
kml: added snap for wp and fence

tridge committed:
asterix: cope with duplicate pkts

tridge committed:
kmlread: report close misses in snap

tridge committed:
misseditor: make 1st column wider

tridge committed:
module: fixed multi-instance unload

tridge committed:
asterix: added playback

tridge committed:
wp: init expected_count

tridge committed:
grapher: use recv_match() with types

tridge committed:
mavmemlog: added recv_match()

tridge committed:
MAVExplorer: use DFReader_binary directly

tridge committed:
MAVExplorer: added progress bar for new bin parser

tridge committed:
MAVExplorer: added messages command

tridge committed:
MAVExplorer: fixed messages rewind

tridge committed:
raise version to 1.6.5

tridge committed:
grapher: save time on date conversion

tridge committed:
grapher: optimise for simple fields

tridge committed:
rline: allow tab completion of expressions for variables

tridge committed:
graph_ui: tabs to spaces

tridge committed:
MAVExplorer: added automatic sync of time axis

tridge committed:
MAVExplorer: improved xaxis sync in graphs

tridge committed:
raise version to 1.6.6

tridge committed:
graph: removed default graph legends

tridge committed:
adjust required pymavlink version

tridge committed:
release 1.6.7

tridge committed:
graph_ui: moved IPC init earlier

tridge committed:
windows: fixed exe build

tridge committed:
raise version to 1.6.8

tridge committed:
wxconsole: catch EOFError

tridge committed:
MAVExplorer: fixed queue crash on MacOS

use pipe instead. Still doesn't do time sync, but doesn't crash

tridge committed:
graph_ui: catch EOFError

tridge committed:
MAVExplorer: make use of mavmmaplog

tridge committed:
prepare for 1.6.9

tridge committed:
MAVExporer: catch child errors

khancyr committed:
modules: reduce calls to get_type()

mavproxy_link: reduce call to get_type()

mavproxy_calibration: reduce call to get_type()

mavproxy_cameraview: reduce call to get_type()

mavproxy_map: reduce call to get_type()

mavproxy_mmap: reduce call to get_type()

mavproxy_misseditor: reduce call to get_type()

khancyr committed:
mavproxy_link: correct membership test

khancyr committed:
optparse_gui: call superclass init

tridge committed:
start on multiproc abstraction

tridge committed:
use multiproc for mavflightview

tridge committed:
map: fixed slipmap on macos

tridge committed:
convert all of MAVProxy to use new multiproc module

tridge committed:
MacOS: added matplotlib for MacOS

fixed phoenix issues

tridge committed:
live_graph: fixed MacOS issues

tridge committed:
live_graph: use local MacOS version of wxagg

tridge committed:
console: fixed background on MacOS

tridge committed:
setup: added MacOS module

tridge committed:
setup: depend on billiard for MacOS

tridge committed:
more multiproc conversion

AndKe committed:
Map: fixed Google Sat map

tridge committed:
misseditor: fixed mission editor to work under MacOS

tridge committed:
map: use newer mouse attributes

fixes an issue with MacOS

tridge committed:
mp_util: fixed marshalling of mouse events

tridge committed:
raise version to 1.7.0

tridge committed:
menu: fixed deprecated warning

tridge committed:
mp_util: catch deprecated attribute warnings

these make no sense for this method

tridge committed:
slipmap: handle zero rotation case

tridge committed:
release version 1.7.1

tridge committed:
increase requirements list for MacOS

tridge committed:
MAVExplorer: added --version option

stephendade committed:
CI - Added Appveyor for building Windows installers

tridge committed:
initial fixes for python3

tridge committed:
grapher: use matplotlib.animation for xlim check timer

tridge committed:
raise version to 1.7.2

tridge committed:
ANUGS: py3 compat

tridge committed:
map: py3 compat

tridge committed:
link: py3 compat

tridge committed:
lib: py3 compat

tridge committed:
mavproxy: py3 compat

tridge committed:
graph: py3 compat

tridge committed:
misseditor: started on py3 compat

tridge committed:
lib: wxsettings conversion to py3

tridge committed:
misseditor: fixed for py3

tridge committed:
horizon: fixed for py3

tridge committed:
fixed mavflightview for py3

tridge committed:
fixes for cuav usage of py3

tridge committed:
suppress some warnings with py3 and phoenix

tridge committed:
fixed error on multi module load

tridge committed:
rline: fixed tab completion in py3

tridge committed:
wp: fixed expected_count error

tridge committed:
misc: fixed remote shell

tridge committed:
layout: fixed on py3

tridge committed:
kml: removed use of StringIO

tridge committed:
release 1.7.3

tridge committed:
reduce CPU usage in graphs

tridge committed:
raise version

khancyr committed:
mavproxy_genobstacles: remove duplicate move() function

khancyr committed:
live_graph_ui: fix drawing

khancyr committed:
mp_menu: fix text dialog for wx4

tridge committed:
MAVExplorer: use the same flightmode colors on all graphs

tridge committed:
MAVExplorer: use fast log loading for *.log

tridge committed:
release 1.7.5

tridge committed:
watch: fixed fwding with no watch set

this broke ardupilot autotest

tridge committed:
release 1.7.6

tridge committed:
adsb: avoid using map when not available

tridge committed:
asterix: fixed on python3

tridge committed:
fixed py3 syntax errors

tridge committed:
param: fixed param download and help for py3

peterbarker committed:
map: fix tile cache so we return cached images

peterbarker committed:
modules: use elif to avoid duplicate string comparisons

tridge committed:
editor: fixed py3 module load

tridge committed:
link: removed time wrap msg

tridge committed:
release 1.7.7

peterbarker committed:
mavproxy_magical: fixes for Python3 fixes

tridge committed:
misc: added devid command

tridge committed:
graphs: added new RC_CHANNELS graph

tridge committed:
misc: added oreoled command

tridge committed:
allow much faster heartbeats

for testing purposes

tridge committed:
rline: fixed tab expansion across underscores

peterbarker committed:
mavproxy_map: correct mouse-rotation division-by-zero flaw

peterbarker committed:
mavproxy_messagerate: simply display an average message rate

peterbarker committed:
mavproxy_example: correct case

peterbarker committed:
mavproxy_genobstacles: add status command, improve help output

peterbarker committed:
mavproxy_adsb: get ElevationMap from relevant map rather than console

the visual console has this attribute, the "SimpleConsole" does not

tridge committed:
joystick: disable verbose debug

tridge committed:
support multiple system ID links

add concept of modules that support multiple vehicles. If they don't
then they only get messages for primary system ID

tridge committed:
menu: prevent invalid menu items being added

tridge committed:
console: added Vehicle menu for selecting vehicle

tridge committed:
console: show system ID in console

tridge committed:
param: auto-switch parameter sets when changing system

tridge committed:
map: display all vehicles on the map

tridge committed:
map: auto-updated fence, wp and rally when switching vehicles

tridge committed:
param: make parameters be component specific

and show components in vehicle menu.

tridge committed:
raise version to 1.8.0

tridge committed:
fixed handling of 2nd GCS on network

tridge committed:
raise version to 1.8.1

tridge committed:
map: fixed multi-vehicle follow

tridge committed:
map: fixed follow bug

thanks to @Aksath-Singhal

tridge committed:
release 1.8.2

tridge committed:
use separate save files for each system

for rally, wp and fence

tridge committed:
dataflash_logger: fixed crash on status

make status not die if not started

peterbarker committed:
messagerate: use floating point arithmetic in status

tridge committed:
heliplane: added status module for heliplanes

tridge committed:
heliplane: fixed typo

tridge committed:
link: fixed showing of console data

stephendade committed:
Layout: Fixed bug in Windows OS

tridge committed:
link: added Hex to list of preferred ports

tridge committed:
added fwdpos setting

for forwarding GLOBAL_POSITION_INT between links

tridge committed:
link: auto-switch link when changing vehicles

tridge committed:
map: fixed multi-vehicle trajectory handling

tridge committed:
prepare for 1.8.3 release

peterbarker committed:
MAVProxy: remove the level command

tridge committed:
added fft to MAVExplorer

tridge committed:
prepare for 1.8.4

tridge committed:
added checkdelay setting

peterbarker committed:
mavproxy_console: correct unknown vehicle type string

peterbarker committed:
mavproxy_console: name components a little better

bugobliterator committed:
live_graph: fix live graph on macos

cclauss committed:
Undefined name: import math for line 79

Discovered via #585.

tridge committed:
added devid command to MAVExplorer

tridge committed:
prepare for 1.8.5 release

stephendade committed:
CI - Added Appveyor Badge

tridge committed:
wp: fixed wp remove bug

mirkix committed:
DGPS: remove double import

rmackay9 committed:
modules: cmdlong velocity mask fix

set additional bits to ignore force, yaw and yaw rate

tridge committed:
link: added Holybro to recognised USB IDs

stephendade committed:
CI: Using Python 3.5 for Windows build

stephendade committed:
Help: Python 3 fixes

stephendade committed:
Map: Python 3 fixes

stephendade committed:
MAVProxy: Python 3 fixes

stephendade committed:
CI: add link to Appveyor badge

tridge committed:
util: added new devid types

stephendade committed:
Windows: Update build file to Python3

tridge committed:
joystick: adjust taranis model

tridge committed:
wp: added slope command

for calculating glide slopes of waypoints

tridge committed:
wp: only do terrain adjustment on relative alt WPs

peterbarker committed:
mavproxy_misc: fix playtune under Python3

tridge committed:
grapher: prevent non-numbers from producing graph errors

peterbarker committed:
MAVExplorer: add option to add title to graph

jaxxzer commented:
:cry:
